source/high-order-euler.cc

```diff
@@ -0,0 +1,698 @@
+ *  \param buffer The buffer that must be filled with u
+ *  \param bstatus The XBraid status structure
+ *  \return Success (0) or failure (1)
+ */
+int
+my_BufPack(braid_App           app,
+           braid_Vector        u,
+           void               *buffer,
+           braid_BufferStatus  bstatus)
+{
+  UNUSED(app);
+  UNUSED(u);
+  UNUSED(buffer);
+  UNUSED(bstatus);
+
+
+//  std::cout << "Buff pack called" << std::endl;
+//  UNUSED(app);
+//  double *dbuffer = (double*)buffer;
+//
+//  unsigned int n_dof = app->vect_size;//number of dofs at fines level
+//  assert(n_dof == u->data.at(0).size() && "the number of degrees of freedom do not match");
+//  unsigned int n_solution_variables = app->TI_p[0]->time_stepping.n_solution_variables;//total problem dimension.(spacedim + density + energy)
+//  unsigned int buf_size = n_dof * n_solution_variables;
+//  dbuffer[0] = buf_size + 1;//buffer + size
+//
+//  for(unsigned int j=0; j < n_solution_variables; ++j)
+//  {
+//    for(unsigned int i=0; i != n_dof; ++i)
+//    {
+//      dbuffer[j*n_dof + i + 1] = (u->data).at(j)(i);
+//    }
+//  }
+//
+//  braid_BufferStatusSetSize(bstatus, (buf_size+1)*sizeof(double));
+//
+  return 0;
+}
+
+/**
+ *  \brief Unpacks a vector that was sent from another processor
+ *
+ *  Unpacks a linear data buffer into the vector pointed to by
+ *  u_ptr.
+ *
+ *  \param app - The braid app struct containing user data
+ *  \param buffer The buffer that must be unpacked
+ *  \param u_ptr The pointer to the vector that is filled
+ *  \param bstatus The XBraid status structure
+ *  \return Success (0) or failure (1)
+ */
+int
+my_BufUnpack(braid_App           app,
+             void               *buffer,
+             braid_Vector       *u_ptr,
+             braid_BufferStatus  bstatus)
+{
+  UNUSED(app);
+  UNUSED(buffer);
+  UNUSED(u_ptr);
+  UNUSED(bstatus);
+
+//  // the vector should be size (dim + 2) X n_dofs at finest level.
+//  std::cout << "buff unpack called\n";
+//  UNUSED(bstatus);
+//  my_Vector *u = NULL; // the vector we will pack the info into
+//  double *dbuffer = (double*)buffer;
+//  int buf_size = static_cast<int>(dbuffer[0]);
+//
+//  u = new(my_Vector);//where does this get deleted?
+//
+//  resizeVector(u->data, app->vect_size, app->n_solution_variables);
+//
+//  //unpack the sent data into the right level
+//  for(int j=0; j < app->n_solution_variables; ++j)
+//  {
+//    for(int i = 0; i < app->vect_size; ++i)
+//    {
+//      u->data.at(j)(i) = dbuffer[j*app->vect_size + i + 1];//+1 because buffer_size = n_dof + 1
+//      assert(j*app->vect_size + i +1 <= buf_size && "somehow, you are exceeding the buffer size as you unpack");
+//    }
+//  }
+//
+//  *u_ptr = u;//modify the u_ptr does this create a memory leak as we just point this pointer somewhere else?
+//
+//  std::cout << "Buffunpack done." << std::endl;
+  return 0;
+}
+
+int main(int argc, char *argv[])
+{
+  MPI_Comm comm = MPI_COMM_WORLD;
+  MPI_Init(&argc, &argv);
```

Comment:
```suggestion
  dealii::Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
```
Instead of using `MPI_Init` and `MPI_Finalize` I suggest to use the scoped object from deal.II.